spec/sec-operations-on-objects-patch.html

```diff
@@ -0,0 +1,25 @@
+<emu-clause id="sec-operations-on-objects">
+  <h1>Operations on Objects</h1>
+  <ins>
+    <emu-clause id="sec-getoption" aoid="GetOption">
+      <h1>GetOption ( _options_, _property_, _type_, _values_, _fallback_ )</h1>
+      <emu-note>This is the same operation as <a href="https://tc39.es/ecma402/#sec-getoption">GetOption</a> in ECMA-402.</emu-note>
+      <p>
+        The abstract operation GetOption extracts the value of the property named _property_ from the provided _options_ object, converts it to the required _type_, checks whether it is one of a List of allowed _values_, and fills in a _fallback_ value if necessary. If _values_ is *undefined*, there is no fixed set of values and any is permitted.
+      </p>
+      <emu-alg>
```

Comment:
```suggestion
      <emu-alg>
        1. Assert: ! IsPropertyKey(_property_) is *true*.
        1. Assert: Type(_type_) is `"boolean"` or `"string"`.
        1. Assert: _possibleValues_ is a List.
```

Reply:
I can add that, but this is a direct copy/paste from ECMA-402

Reply:
402 shouldn’t constrain 262 tho; i think that we should get this op right and 402 can adapt.

Reply:
Also, `Get` already asserts `IsPropertyKey` as well as that `O` is an Object.

Reply:
The ECMAScript spec does not use the `!` shorthand for `IsPropertyKey`.

Reply:
Every abstract operation call should use ! or ? (when not inspecting a returned completion result).

Fair point that `Get` already does the assertions; I still find it clearer to repeat them here tho, so I don't have to click through to Get to see that.